### PR TITLE
Move Configuration to compas

### PR DESCRIPTION
**Pull request by beverlylytle**
I leave this in draft form until the corresponding compas pr is merged and released.  But I do wonder about how to document the change.  I've removed `Configuration` from the docs, but there's still a way to import it through `compas_fab.robots.Configuration`.  How should that be reflected in the docs and the changelog?

### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.rst` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [x] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas_fab.robots.Configuration`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by beverlylytle** [comment]
Also, to be quite honest, I did not check that each and every one of the ghx still works...

**Comment by gonzalocasas** [comment]
I think we're missing an alias on `compas_fab.robots` so that code can still work for a little while.

**Comment by beverlylytle** [comment]
Sorry, I don't understand.  In `src/compas_fab/robots/configuration.py` `Configuration` is imported from `compas.robots` and then listed in `__all__`.  Is that not sufficient?  It seems to me with that `from compas_fab.robots import Configuration` still works.

**Comment by gonzalocasas** [comment]
And now compas requirement can be bumped to 1.3

**Comment by gonzalocasas** [comment]
> Sorry, I don't understand. In `src/compas_fab/robots/configuration.py` `Configuration` is imported from `compas.robots` and then listed in `__all__`. Is that not sufficient? It seems to me with that `from compas_fab.robots import Configuration` still works.

sorry! didn't see it (I thought the file was deleted, not 98% deletions :P )

**Comment by gonzalocasas** [comment]
This looks pretty much ready to go, right?

**Comment by beverlylytle** [comment]
Code-wise, yes.  But what about documentation/changelog-wise? How do we announce that `compas_fab.robots.Configuration` is deprecated in favor of `compas.robots.Configuration`, but both still work?

File: CHANGELOG.rst [hunk 1]
```diff
@@ -16,6 +16,8 @@ Unreleased
 
 **Changed**
 
+* The `Configuration` class has moved to `compas.robots`, but is still aliased within `compas_fab.robots`
```
Comment (gonzalocasas) [inline CHANGELOG.rst:19]: We could add an extra line in the `Deprecated` section of the changelog.

**Review by gonzalocasas — APPROVED** [review]
LGTM!

Question before merge: should we release an intermediate version that has the changes to configuration-as-dict beforehand to have an intermediate update path?

**Comment by beverlylytle** [comment]
> Question before merge: should we release an intermediate version that has the changes to configuration-as-dict beforehand to have an intermediate update path?

I don't have a strong opinion about this.

**Comment by gonzalocasas** [comment]
> I don't have a strong opinion about this.

Then let's go ahead and merge it :)

**Pull request merged**